geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/RichCountMinSketch.scala

```diff
@@ -0,0 +1,108 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.utils.stats
+
+import com.clearspring.analytics.stream.frequency.CountMinSketch
+import RichCountMinSketch._
+/**
+  * Extends counts min sketch with public access to protected variables for more efficient operations.
+  */
+class RichCountMinSketch(val sketch: CountMinSketch) extends AnyVal {
+
+  def width = refWidth(sketch)
+  def depth = refDepth(sketch)
+
+  def table: Array[Array[Long]] = refTable(sketch)
+  def setSize(size: Long): Unit = refSetSize(sketch, size)
+
+  def add(other: CountMinSketch): Unit = {
+    // note: we assume that seed is equal
+    if (depth != refDepth(other) || width != refWidth(other)) {
+      throw new IllegalArgumentException("Can't merge CountMinSketch of different sizes")
+    }
+    val t = table
+    var i = 0
+    while (i < t.length) {
+      val row = t(i)
+      val otherRow = refTable(other)(i)
+      var j = 0
+      while (j < row.length) {
+        row(j) += otherRow(j)
+        j += 1
+      }
+      i += 1
+    }
+    setSize(sketch.size + other.size)
+  }
+
+  def clear(): Unit = {
+    var i = 0
+    while (i < depth) {
+      val row = table(i)
+      var j = 0
+      while (j < width) {
+        row(j) = 0L
+        j += 1
+      }
+      i += 1
+    }
+    setSize(0L)
+  }
+
+  def isEquivalent(other: CountMinSketch): Boolean = {
+    if (sketch.size != other.size || depth != refDepth(other) || width != refWidth(other)) {
+      return false
+    }
+    var i = 0
+    while (i < depth) {
+      if (refHashA(sketch)(i) != refHashA(other)(i)) {
+        return false
+      }
+      val t = table
+      val row = table(i)
+      val otherRow = refTable(other)(i)
```

Comment:
since these are all declaredFields, I'm guessing repeated calling doesn't really have a performance penalty? or would it be better to call once and store to a local val?